ext/zend_test/test.c

```diff
@@ -364,6 +365,67 @@ static ZEND_FUNCTION(zend_test_crash)
 	php_printf("%s", invalid);
 }
 
+zend_mm_heap* zend_test_heap;
+zend_mm_heap* zend_orig_heap;
+volatile uint32_t lineno = 0;
+
+static bool has_opline(zend_execute_data *execute_data)
+{
+	return execute_data
+		&& execute_data->func
+		&& ZEND_USER_CODE(execute_data->func->type)
+		&& execute_data->opline
+	;
+}
+
+#pragma GCC push_options
+#pragma GCC optimize("O0")
```

Comment:
Why is it necessary to disable optimizations (same for other functions)?

Reply:
I am not really doing anything with the line number, just storing it without any effect. A compiler could optimise that away and not do anything anymore in this function.
But @bwoebi pointed out, that I could use `assert()` for this case as well, which made it possible to remove all the compiler hints.